### PR TITLE
Bug Fix: `compute_loss` returns vector instead of scalar

### PR DESCRIPTION
**Pull request by ayusbhar2**

### CODE REVIEW

File: BasicNNMatrix.py [hunk 1]
```diff
@@ -88,31 +60,31 @@ def compute_weight_multiplication(weight_list):
...
     x = y-z
-    print(f"y is: {y}")
-    print(f"z is: {z}")
-    print(f"x is: {x}")
+    # print(f"y is: {y}")
+    # print(f"z is: {z}")
+    # print(f"x is: {x}")
     squared_vector = np.square(x)
-    print(f"squared error is: {squared_vector}")
-    loss = squared_vector.sum(axis=1)
```
Comment (ayusbhar2) [inline BasicNNMatrix.py:107]: `axis=1` was the problem. A `1-D` numpy array does not have axis 1. It only has axis 0. In fact for a 1 D array we don't need to specify the axis to the `sum` function.
